### PR TITLE
Fix null item in sendEquipmentChange

### PR DESCRIPTION
**Pull request by SplotyCode**

### CODE REVIEW

**Comment by SplotyCode** [comment]
Should i create a new paper patch instead of a fixup?

**Comment by Leguan16** [comment]
This looks like its a new patch. 
For now, don't use imports, but FQN instead.

**Comment by SplotyCode** [comment]
> This looks like its a new patch. For now, don't use imports, but FQN instead.

Ah sorry now that you mention it males total since as it would be annoying when resolving upstream conflicts

**Review by Machine-Maker — APPROVED** [review]

File: patches/server/1040-Fix-null-item-for-Player-sendEquipmentChange.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 604ed1f6742a6b028b9db2809f7bd4b9a3b38f4d..96ab6b5787839bfb6be682bdbc29a011eab8384c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1072,7 +1072,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+ 
+     @Override
+     public void sendEquipmentChange(LivingEntity entity, EquipmentSlot slot, ItemStack item) {
+-        this.sendEquipmentChange(entity, Map.of(slot, item));
++        this.sendEquipmentChange(entity, java.util.Collections.singletonMap(slot, item)); // Paper - replace Map.of
```
Comment (MiniDigger) [inline patches/server/1040-Fix-null-item-for-Player-sendEquipmentChange.patch]:
```suggestion
+        this.sendEquipmentChange(entity, java.util.Collections.singletonMap(slot, item)); // Paper - replace Map.of to allow null values
```
maybe a bit more explicit?

Comment (MiniDigger) [reply patches/server/1040-Fix-null-item-for-Player-sendEquipmentChange.patch]: maybe change the patch title to into allow null items for ...

**Review by lynxplay — APPROVED** [review]

**Pull request merged**